contracts/lotterybook/drawer_test.go

```diff
@@ -0,0 +1,376 @@
+		amount = append(amount, 128)
+	}
+	id, err := drawer.createLottery(context.Background(), receiver, amount, 10086, nil, nil)
+	if err != nil {
+		t.Fatalf("Failed to create lottery with lots of receivers")
+	}
+	var cheques []*Cheque
+	var receivers []common.Address
+	drawer.cdb.listCheques(drawer.address, func(addr common.Address, hash common.Hash, cheque *Cheque) bool {
+		if id == hash {
+			cheques = append(cheques, cheque)
+			receivers = append(receivers, addr)
+		}
+		return true
+	})
+	lottery := drawer.cdb.readLottery(drawer.address, id)
+	if len(lottery.Receivers) != len(cheques) {
+		t.Fatalf("The receiver number is different")
+	}
+	for _, r := range lottery.Receivers {
+		var find bool
+		for _, rr := range receivers {
+			if r == rr {
+				find = true
+				break
+			}
+		}
+		if !find {
+			t.Fatalf("Receiver mismatch")
+		}
+	}
+}
+
+func TestIssueCheque(t *testing.T) {
+	env := newTestEnv(t)
+	defer env.close()
+
+	var exit = make(chan struct{})
+	defer close(exit)
+
+	// Start the automatic blockchain.
+	go func() {
+		ticker := time.NewTicker(time.Millisecond * 10)
+		for {
+			select {
+			case <-ticker.C:
+				env.backend.Commit()
+			case <-exit:
+				return
+			}
+		}
+	}()
+	drawer, err := NewChequeDrawer(env.drawerAddr, env.contractAddr, bind.NewKeyedTransactor(env.drawerKey), nil, env.backend.Blockchain(), env.backend, env.backend, env.drawerDb)
+	if err != nil {
+		t.Fatalf("Faield to create drawer, err: %v", err)
+	}
+	defer drawer.Close()
+	drawer.keySigner = func(data []byte) ([]byte, error) {
+		sig, _ := crypto.Sign(data, env.drawerKey)
+		return sig, nil
+	}
+	id, err := drawer.createLottery(context.Background(), []common.Address{env.draweeAddr}, []uint64{128}, 10086, nil, nil)
+	if err != nil {
+		t.Fatalf("Faield to create lottery, err: %v", err)
+	}
+	id2, err := drawer.createLottery(context.Background(), []common.Address{env.draweeAddr, env.drawerAddr}, []uint64{40, 60}, 10086, nil, nil)
+	if err != nil {
+		t.Fatalf("Faield to create lottery, err: %v", err)
+	}
+	id3, err := drawer.createLottery(context.Background(), []common.Address{env.draweeAddr, env.drawerAddr}, []uint64{40, 45}, 10086, nil, nil)
+	if err != nil {
+		t.Fatalf("Faield to create lottery, err: %v", err)
+	}
+	var cases = []struct {
+		payer     common.Address
+		id        common.Hash
+		amount    uint64
+		expectErr bool
+	}{
+		// Lottery 1
+		{env.draweeAddr, id, 0, true},
+		{env.draweeAddr, id, 32, false},
+		{env.draweeAddr, id, 96, false},
+		{env.draweeAddr, id, 32, true},
+		// Lottery 2, each one has 50 assigned
+		{env.draweeAddr, id2, 25, false},
+		{env.draweeAddr, id2, 25, false},
+		{env.draweeAddr, id2, 1, true},
+		// Lottery 3, each one has 42.5 assigned
+		{env.draweeAddr, id3, 21, false},
+		{env.draweeAddr, id3, 21, false},
+		{env.draweeAddr, id3, 1, true},
+	}
+	for index, c := range cases {
+		_, err := drawer.issueCheque(c.payer, c.id, c.amount, true)
```

Comment:
I replaced this line with the following (to test `IssueCheque` which is currently not  tested)
```suggestion
		_, err := drawer.IssueCheque(c.payer, c.amount)
```
This failed with the following error:
`Case 3 expect error, got nil`